### DISCUSSION TITLE
Cleaner way to negate last column?

### DISCUSSION THREAD

**Discussion by joezappie** [josdejong/mathjs#3161] [Q&A]
I'm just curious as theres a lot to this library I dont know yet, is there a simpler way to negate the last column in an arbitrarily sized matrix?

```
    const [rows, columns] = math.size(matrix);
    const index = math.index(math.range(0, rows), columns - 1);
    const v = math.subset(matrix, index);
    matrix = math.subset(matrix, index, math.multiply(v, -1));
```

**Comment by josdejong** [comment]
You solution is a good and clear one. 

Other ideas:

- You could use the `.map()` method, and in the callback check whether it is a value from the last column, and if so return it negated, and otherwise return the value itself.
- You can use the expression parser for a more concise notation, see docs: https://mathjs.org/docs/expressions/syntax.html#matrices

**Comment by dvd101x** [comment]
Following Jos's advise to use the expression parser. You could try
```js
const scope = {M: matrix}
math.evaluate("M[:, end] = -M[:, end]", scope)
matrix = scope.M
```
